examples/kdbget.c

```diff
@@ -8,12 +9,52 @@ int main()
 	KDB *handle = kdbOpen(key);
 
 	kdbGet(handle, myConfig, key);
-	// TODO: check for errors and warnings in key
 
-	keyDel(key);
+	//check for error
+	if(keyGetMeta (key,"error"))
+	{
+		printf("Error occurred: %s\n",
+				keyString(keyGetMeta(key,"error/description")));
+		//fields for more information are listed in the value from
+		//the Key returned by keyGetMeta(key,"error")
+	}
+	//check for warnings
+	if(keyGetMeta (key,"warnings"))
+	{
+		const char * warning_count = keyString (keyGetMeta (key,"warnings"));
+		char buffer[] = "warnings/#00/description";
+
+		do {
+			const Key *warnkey = keyGetMeta (key,buffer);
+			printf ("Warning occurred: %s\n",keyString (warnkey));
+			//fields for more information are listed in the value from
+			//the Key returned by keyGetMeta(key,"warnings/#00")
+			buffer[11]++;
+			if (buffer[11] > '9')
+			{
+				buffer[11] = '0';
+				buffer[10]++;
+			}
+		} while (strncmp (&buffer[10],warning_count,2)<=0);
+	}
+
+	keyDel (key);
+
+	//insert a key so lookup has something to lookup
```

Comment:
rephrase e.g.: "lookup always will find something, even if kdbGet() did not return the value"

the appending can also be done before the kdbGet(), then a user supplied configuration would win and make the program execution of the example more interesting